lib/cli.rb

```diff
@@ -587,7 +591,83 @@ def self.check_port_validity(port)
     end
   end
 
+  desc "Inspect container image"
+  long_desc <<-LONGDESC
+    Inspect container image and generate system descripton from inspected data.
+
+    Multiple scopes can be passed as comma-separated list. If no specific scopes
+    are given, all scopes are inspected.
+
+    Available scopes: #{AVAILABLE_SCOPE_LIST}
+  LONGDESC
+  arg "IMAGENAME"
+  command "inspect-container" do |c|
+    supports_filtering(c)
+    define_inspect_command_options(c)
+    c.switch ["docker", :d], required: true, negatable: false,
+      desc: "Inspect a docker container"
+
+    c.action do |_global_options,options,args|
+      image = shift_arg(args, "IMAGENAME")
+      system = DockerSystem.new(image)
+      inspector_task = InspectTask.new
+      scope_list = process_scope_option(options[:scope], options["exclude-scope"])
+      name = options[:name] || image
+
+      if !scope_list.empty?
+        inspected_scopes = " for #{Machinery::Ui.internal_scope_list_to_string(scope_list)}"
+      end
+      Machinery::Ui.puts "Inspecting #{image}#{inspected_scopes}..."
+
+      inspect_options = {}
+      if options["show"]
+        inspect_options[:show] = true
+      end
+      if options["verbose"]
+        inspect_options[:verbose] = true
+      end
+      if options["extract-files"] || options["extract-changed-config-files"]
+        inspect_options[:extract_changed_config_files] = true
+      end
+      if options["extract-files"] || options["extract-changed-managed-files"]
+        inspect_options[:extract_changed_managed_files] = true
+      end
+      if options["extract-files"] || options["extract-unmanaged-files"]
+        inspect_options[:extract_unmanaged_files] = true
+      end
+
+      filter = FilterOptionParser.parse("inspect", options)
+
+      if options["verbose"] && !filter.empty?
+        Machinery::Ui.puts "\nThe following filters are applied during inspection:"
+        Machinery::Ui.puts filter.to_array.join("\n") + "\n\n"
+      else
+        show_filter_note(scope_list, filter)
+      end
 
+      begin
+        system.start
+        inspector_task.inspect_system(
+          system_description_store,
+          system,
+          name,
+          CurrentUser.new,
+          scope_list,
+          filter,
+          inspect_options
+        )
+      ensure
+        system.stop
+      end
+
+
+      Hint.print(:show_data, name: name)
+
+      if !options["extract-files"] || Inspector.all_scopes.count != scope_list.count
+        Hint.print(:do_complete_inspection, name: name, docker_container: image)
+      end
+    end
+  end
```

Comment:
There is a lot of duplication between the code handling the "inspect" command and the code doing "inspect-container". We should dry that up.